### PR TITLE
fix: Pattern.record

### PR DESCRIPTION
**Pull request by jamiehodge**
## Purpose and Motivation

Fixes https://github.com/supercollider/supercollider/issues/5879

## Types of changes

- Bug fix

## To-do list

- [x] Code is tested (locally)
- [ ] All tests are passing
- [ ] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

**Review by telephon — COMMENTED** [review]
Looks good!

You should leave out the semicolon within the function.


You could also write it in short:
```
recHeaderFormat = recHeaderFormat ?? { server.recHeaderFormat };
```

but that is a matter of style.

**Comment by jamiehodge** [comment]
Hi Julian! Long time no see.

Updated, hopefully correctly.

File: SCClassLibrary/Common/Streams/Patterns.sc [hunk 1]
```diff
@@ -101,6 +101,9 @@ Pattern : AbstractFunction {
 		var buffer, nodeID, defname;
 		var pattern = if(dur.notNil) { Pfindur(dur, this) } { this };
 
+        headerFormat = headerFormat ?? { server.recHeaderFormat };
+		sampleFormat = sampleFormat ?? { server.recSampleFormat };
```
Comment (jrsurge) [inline SCClassLibrary/Common/Streams/Patterns.sc:105]: is there an advantage here of `??` over `?`? these operators always confuse me, trying to learn how/when is best to pick one over the other - any advice greatly appreciated!

Comment (dyfer) [reply SCClassLibrary/Common/Streams/Patterns.sc:105]: I think ?? evaluates the function on the right, while ? doesn't, just using that value directly. I might be wrong though.

Comment (telephon) [reply SCClassLibrary/Common/Streams/Patterns.sc:105]: In `?`, the call `server.recSampleFormat` is always made, even if it isn't needed. With `??`, it is only made if `headerFormat` is nil.

**Comment by dyfer** [comment]
@telephon do you think this good to go?

File: SCClassLibrary/Common/Streams/Patterns.sc (hunk 1 above)
Comment (jrsurge) [reply SCClassLibrary/Common/Streams/Patterns.sc:105]: Thanks, thought it might be something like that!

**Review by telephon — APPROVED** [review]

**Pull request merged**